docs/sphinx/pypi-dependencies.md

```diff
@@ -0,0 +1,153 @@
+interpreter used to run `py_binary` targets. By default, `pip_parse` uses the
+system command `"python3"`. To override this, pass in the `python_interpreter`
+attribute or `python_interpreter_target` attribute to `pip_parse`.
+
+You can have multiple `pip_parse`s in the same workspace.  Or use the pip
+extension multiple times when using bzlmod. This configuration will create
+multiple external repos that have no relation to one another and may result in
+downloading the same wheels numerous times.
+
+As with any repository rule, if you would like to ensure that `pip_parse` is
+re-executed to pick up a non-hermetic change to your environment (e.g., updating
+your system `python` interpreter), you can force it to re-execute by running
+`bazel sync --only [pip_parse name]`.
+
+:::{note}
+The `pip_install` rule is deprecated. `pip_parse` offers identical
+functionality, and both `pip_install` and `pip_parse` now have the same
+implementation. The name `pip_install` may be removed in a future version of the
+rules.
+:::
+
+The maintainers have made all reasonable efforts to facilitate a smooth
+transition. Still, some users of `pip_install` will need to replace their
+existing `requirements.txt` with a fully resolved set of dependencies using a
+tool such as `pip-tools` or the `compile_pip_requirements` repository rule.
+
+### Using third_party packages as dependencies
+
+Each extracted wheel repo contains a `py_library` target representing
+the wheel's contents. There are two ways to access this library. The
+first uses the `requirement()` function defined in the central
+repo's `//:requirements.bzl` file. This function maps a pip package
+name to a label:
+
+```
+load("@my_deps//:requirements.bzl", "requirement")
+
+py_library(
+    name = "mylib",
+    srcs = ["mylib.py"],
+    deps = [
+        ":myotherlib",
+        requirement("some_pip_dep"),
+        requirement("another_pip_dep"),
+    ]
+)
+```
+
+The reason `requirement()` exists is that the pattern for the labels,
+while not expected to change frequently, is not guaranteed to be
+stable. Using `requirement()` ensures you do not have to refactor
+your `BUILD` files if the pattern changes.
+
+On the other hand, using `requirement()` has several drawbacks; see
+[this issue][requirements-drawbacks] for an enumeration. If you don't
+want to use `requirement()`, you can use the library
+labels directly instead. For `pip_parse`, the labels are of the following form:
+
+```
+@{name}_{package}//:pkg
+```
+
+Here `name` is the `name` attribute that was passed to `pip_parse` and
+`package` is the pip package name with characters that are illegal in
+Bazel label names (e.g. `-`, `.`) replaced with `_`. If you need to
+update `name` from "old" to "new", then you can run the following
+buildozer command:
+
+```shell
+buildozer 'substitute deps @old_([^/]+)//:pkg @new_${1}//:pkg' //...:*
+```
+
+For `pip_install`, the labels are instead of the form:
+
+```
+@{name}//pypi__{package}
```

Comment:
nit: it would be good to remove this, because no code is producing such labels anymore, I think.

Reply:
Done